### PR TITLE
Sligh improvement of UrlPattern

### PR DESCRIPTION
**Pull request by simonbrandhof**
Sligh improvement of UrlPattern

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/authentication/ValidateJwtTokenFilter.java [hunk 1]
```diff
@@ -44,8 +44,8 @@ public ValidateJwtTokenFilter(JwtHttpHandler jwtHttpHandler) {
   @Override
   public UrlPattern doGetPattern() {
     return UrlPattern.builder()
-      .setIncludePatterns("/*")
-      .setExcludePatterns("/css/*", "/fonts/*", "/images/*", "/js/*", "/static/*")
+      .includes("/*")
+      .excludes(UrlPattern.Builder.staticResourcePatterns())
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/authentication/ValidateJwtTokenFilter.java]: Good changes !

File: server/sonar-server/src/test/java/org/sonar/server/authentication/ValidateJwtTokenFilterTest.java [hunk 2]
```diff
@@ -52,8 +52,12 @@ public void setUp() throws Exception {
...
   @Test
   public void do_get_pattern() throws Exception {
-    assertThat(underTest.doGetPattern().getIncludePatterns()).containsOnly("/*");
-    assertThat(underTest.doGetPattern().getExcludePatterns()).containsOnly("/css/*", "/fonts/*", "/images/*", "/js/*", "/static/*");
+    assertThat(underTest.doGetPattern().matches("/")).isTrue();
+    assertThat(underTest.doGetPattern().matches("/foo")).isTrue();
+
+    // exclude static resources
+    assertThat(underTest.doGetPattern().matches("/css/style.css")).isFalse();
+    assertThat(underTest.doGetPattern().matches("/images/logo.png")).isFalse();
```
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/authentication/ValidateJwtTokenFilterTest.java]: Maybe tests on fonts and js should be added

Comment (simonbrandhof) [inline server/sonar-server/src/test/java/org/sonar/server/authentication/ValidateJwtTokenFilterTest.java]: good point

File: server/sonar-server/src/main/java/org/sonar/server/authentication/ValidateJwtTokenFilter.java (hunk 1 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/authentication/ValidateJwtTokenFilter.java]: But please use static import for staticResourcePatterns()

File: sonar-plugin-api/src/main/java/org/sonar/api/web/ServletFilter.java [hunk 3]
```diff
@@ -50,141 +49,100 @@ public UrlPattern doGetPattern() {
...
      */
     @Deprecated
-    public String getUrl(){
-      // Before 6.0, it was only possible to include one url
-      if (excludePatterns.isEmpty() && includePatterns.size() == 1 ) {
-        return includePatterns.get(0).getUrl();
-      }
-      throw new IllegalStateException("this method is deprecated and should not be used anymore");
+    public String getUrl() {
+      return "";
```
Comment (julienlancelot) [inline sonar-plugin-api/src/main/java/org/sonar/api/web/ServletFilter.java]: As discussed, you should return value if only one included pattern exists in order to not break backward compatibility

**Comment by julienlancelot** [comment]
Very good improvement and smart usage of Predicate, good job !
LGTM

**Pull request closed without merging**